optuna_dashboard/ts/components/DataGrid.tsx

```diff
@@ -84,6 +86,41 @@ function DataGrid<T>(props: {
     setPage(0)
   }
 
+  const PaginationTextFieldComponent: React.FC<{
+    onPageNumberSubmit: (value: number) => void
+    maxPageNumber: number
+  }> = ({ onPageNumberSubmit, maxPageNumber }) => {
+    // This component is separated from DataGrid to prevent `DataGrid` from re-rendering the page,
+    // every time any letters are input.
+    const [specifiedPageText, setSpecifiedPageText] = React.useState("")
+
+    const handleSubmitPageNumber = (
+      event: React.FormEvent<HTMLFormElement>
+    ) => {
+      event.preventDefault()
+      const newPage = parseInt(specifiedPageText, 10)
+      setSpecifiedPageText("") // reset the input field
```

Comment:
Sorry for my poor explanation. The `setSpecifiedPageText("")` should be the procedure if the process is correctly handled. In that sense, this line should be put on the last line. More detail, the reason is that if we use early return e.g. `if (Number.isNaN(newPage)) {`, we should keep text in the field because the user can fix it later.

Reply:
Thank you for the clarification! Your explanation actually makes sense")